src/migrations/db_migration.rs

```diff
@@ -0,0 +1,123 @@
+use std::collections::{BTreeSet, BTreeMap};
+use indexmap::IndexMap;
+
+use crate::connect::Connection;
+
+#[derive(Debug, Clone, edgedb_tokio::Queryable)]
+// TODO(tailhook) this has to be open-ended enumeration
+pub(crate) enum MigrationGeneratedBy {
+    DevMode,
+    DDLStatement,
+}
+
+pub(crate) trait SortableMigration {
```

Comment:
I think we don't need this trait any more as we are merging the impls - using the struct directly in `linearize_db_migrations()` would be enough.

Reply:
I decided to leave it in because I plan further work to unify FS migrations with DB migrations at the trait level.